cmd/minishift/cmd/start_preflight.go

```diff
@@ -374,6 +388,22 @@ func checkHypervDriverUser() bool {
 	return true
 }
 
+// checkXhyveInstalled returns true if XHyve is installed
+func checkXhyveInstalled() bool {
+	path, err := exec.LookPath("xhyve")
+	if err != nil {
+		return false
+	}
+	fi, _ := os.Stat(path)
+	if fi.Mode()&os.ModeSymlink != 0 {
+		path, err = os.Readlink(path)
+		if err != nil {
+			return false
+		}
+	}
+	return true
```

Comment:
-1, We do not need Xhyve to be installed on macOS if we have the driver installed as the driver contains the Xhyve also.

Reply:
Yes i didn't know about it, we don't need this check.

Reply:
We would need a check to see if the platform has supported for running a hypervisor... This is the HypervisorKit or similar...

Reply:
To check if the platform supports VM driver we have a check already here, https://github.com/minishift/minishift/blob/84bbae500a15995e16843e22f67ef1d6c203424e/cmd/minishift/cmd/start_preflight.go#L53-L58

Reply:
you need to check for the existence of _docker-machine-driver-xhyve_ on the path. That's the actual driver which needs to be installed manually and which needs to have the suid bit set.